diff --git a.bender/git/checkouts/cva6-7b729dce775b20ea/core/cva6.sv b.bender/git/checkouts/cva6-7b729dce775b20ea/core/cva6.sv
index 12684f1e..59e2824c 100644
--- a/core/cva6.sv
+++ b/core/cva6.sv
@@ -1449,6 +1449,10 @@ module cva6
         CVA6Cfg.DCacheType == config_pkg::HPDCACHE_WT_WB
   )
   begin : gen_cache_hpd
+
+    amo_req_t  amo_req_masked;
+    amo_resp_t amo_resp_precut;
+
     cva6_hpdcache_subsystem #(
         .CVA6Cfg   (CVA6Cfg),
         .icache_areq_t(icache_areq_t),
@@ -1486,8 +1490,8 @@ module cva6
         .dcache_flush_ack_o(dcache_flush_ack_cache_ctrl),
         .dcache_miss_o     (dcache_miss_cache_perf),
 
-        .dcache_amo_req_i (amo_req),
-        .dcache_amo_resp_o(amo_resp),
+        .dcache_amo_req_i (amo_req_masked),
+        .dcache_amo_resp_o(amo_resp_precut),
 
         .dcache_cmo_req_i ('0  /*FIXME*/),
         .dcache_cmo_resp_o(  /*FIXME*/),
@@ -1514,6 +1518,41 @@ module cva6
     );
     assign inval_ready   = 1'b1;
     assign miss_vld_bits = '0;
+
+    always_comb begin
+      amo_req_masked = amo_req;
+      amo_req_masked.req = amo_req.req & ~amo_resp.ack;
+    end
+
+    // a spill register for amo_resp, for timing optimization
+    spill_register #(
+        .T     (logic [$bits(amo_resp_precut.result)-1:0]),
+        .Bypass(0)
+    ) i_amo_resp_result (
+        .clk_i  (clk_i),
+        .rst_ni (rst_ni),
+        .valid_i(amo_resp_precut.ack),
+        .ready_o(),
+        .data_i (amo_resp_precut.result),
+        .valid_o(),
+        .ready_i(1'b1),
+        .data_o (amo_resp.result)
+    );
+
+    spill_register #(
+        .T     (logic),
+        .Bypass(0)
+    ) i_amo_resp_ack (
+        .clk_i  (clk_i),
+        .rst_ni (rst_ni),
+        .valid_i(1'b1),
+        .ready_o(),
+        .data_i (amo_resp_precut.ack),
+        .valid_o(),
+        .ready_i(1'b1),
+        .data_o (amo_resp.ack)
+    );
+
   end else begin : gen_cache_wb
     std_cache_subsystem #(
         // note: this only works with one cacheable region
