diff --git a.bender/git/checkouts/pulp-c910-f7479e82489da398/vendor/thead_openc910/C910_RTL_FACTORY/gen_rtl/lsu/rtl/ct_lsu_ctrl.v b.bender/git/checkouts/pulp-c910-f7479e82489da398/vendor/thead_openc910/C910_RTL_FACTORY/gen_rtl/lsu/rtl/ct_lsu_ctrl.v
index 6b76daf..1938767 100644
--- a/vendor/thead_openc910/C910_RTL_FACTORY/gen_rtl/lsu/rtl/ct_lsu_ctrl.v
+++ b/vendor/thead_openc910/C910_RTL_FACTORY/gen_rtl/lsu/rtl/ct_lsu_ctrl.v
@@ -280,8 +280,8 @@ input   [2  :0]  lsu_had_icc_state;
 input   [7  :0]  lsu_had_lfb_addr_entry_dcache_hit; 
 input   [7  :0]  lsu_had_lfb_addr_entry_rcl_done;  
 input   [7  :0]  lsu_had_lfb_addr_entry_vld;       
-input   [1  :0]  lsu_had_lfb_data_entry_last;      
-input   [1  :0]  lsu_had_lfb_data_entry_vld;       
+input   [LFB_DATA_ENTRY-1  :0]  lsu_had_lfb_data_entry_last;      
+input   [LFB_DATA_ENTRY-1  :0]  lsu_had_lfb_data_entry_vld;       
 input            lsu_had_lfb_lf_sm_vld;            
 input   [12 :0]  lsu_had_lfb_wakeup_queue;         
 input   [2  :0]  lsu_had_lm_state;                 
