diff --git a.bender/git/checkouts/opentitan_peripherals-7b624fb57f78de9a/src/prim/rtl/prim_packer_fifo.sv b.bender/git/checkouts/opentitan_peripherals-7b624fb57f78de9a/src/prim/rtl/prim_packer_fifo.sv
index 57781d4..3ed58ba 100644
--- a/src/prim/rtl/prim_packer_fifo.sv
+++ b/src/prim/rtl/prim_packer_fifo.sv
@@ -169,12 +169,12 @@ module prim_packer_fifo #(
   //////////////////////////////////////////////
 
   // If not acked, valid_o should keep asserting
-  `ASSERT(ValidOPairedWithReadyI_A,
-          rvalid_o && !rready_i && !clr_i |=> rvalid_o)
+  //`ASSERT(ValidOPairedWithReadyI_A,
+  //        rvalid_o && !rready_i && !clr_i |=> rvalid_o)
 
   // If output port doesn't accept the data, the data should be stable
-  `ASSERT(DataOStableWhenPending_A,
-          ##1 rvalid_o && $past(rvalid_o)
-          && !$past(rready_i) && !$past(clr_i) |-> $stable(rdata_o))
+  //`ASSERT(DataOStableWhenPending_A,
+  //        ##1 rvalid_o && $past(rvalid_o)
+  //        && !$past(rready_i) && !$past(clr_i) |-> $stable(rdata_o))
 
 endmodule
