diff --git a.bender/git/checkouts/cva6-7b729dce775b20ea/core/cva6_fifo_v3.sv b.bender/git/checkouts/cva6-7b729dce775b20ea/core/cva6_fifo_v3.sv
index 631ce226..4edddd38 100644
--- a/core/cva6_fifo_v3.sv
+++ b/core/cva6_fifo_v3.sv
@@ -117,8 +117,10 @@ module cva6_fifo_v3 #(
     end
 
     if (pop_i && ~empty_o) begin
-      data_ft_n = data_i;
-      if (FPGA_EN && FPGA_ALTERA) first_word_n = first_word_q && push_i;
+      if (FPGA_EN && FPGA_ALTERA) begin
+        data_ft_n = data_i;
+        first_word_n = first_word_q && push_i;
+      end
       // read from the queue is a default assignment
       // but increment the read pointer...
       if (read_pointer_n == FifoDepth[ADDR_DEPTH-1:0] - 1) read_pointer_n = '0;
